### PR TITLE
Bug-compatible refactor of docker-env + output tests

### LINKED ISSUE #6524: docker-env command does not work in powershell (v1.7)
**Environment**
- minikube v1.7.1 on Microsoft Windows 10 Pro 10.0.18362 Build 18362

**Commnd Execute result**
```
PS C:\Windows\system32> minikube docker-env --shell powershell | Invoke-Expression
Invoke-Expression : 引数が空の文字列であるため、パラメーター 'Command' にバインドできません。
発生場所 行:1 文字:42
+ minikube docker-env --shell powershell | Invoke-Expression
+                                          ~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:PSObject) [Invoke-Expression]、ParameterBindingValidationException
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorEmptyStringNotAllowed,Microsoft.PowerShell.Commands.InvokeExpressionCommand
```

*Remarks*
Line break position is different between v1.7.2 and v1.6.2.
There is a line break after $Env.
<details>
  <summary>v1.6.2</summary>

```
PS C:\Windows\system32> docker-env --shell powershell
$Env:DOCKER_TLS_VERIFY = "1"
$Env:DOCKER_HOST = "tcp://192.168.163.55:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\***\.minikube\certs"
# Run this command to configure your shell:
# & minikube docker-env | Invoke-Expression

PS C:\Windows\system32>
```
</details>

<details>
  <summary>v1.7.2</summary>

```
PS C:\Windows\system32> minikube docker-env --shell powershell
$Env:DOCKER_TLS_VERIFY = "1"
$Env:DOCKER_HOST = "tcp://192.168.163.55:2376"
$Env:DOCKER_CERT_PATH = "C:\Users\***\.minikube\certs"
$Env:MINIKUBE_ACTIVE_DOCKERD = "minikube"

# Please run command bellow to point your shell to minikube's docker-daemon :
# & minikube -p minikube docker-env | Invoke-Expression

PS C:\Windows\system32>
```
</details>

### PR DESCRIPTION
**Pull request by tstromberg**
The previous tests for `docker-env` did not provide much value, causing issues like #6524 to be both easy to introduce, and difficult to confidently fix.

This PR refactors `docker-env` to be trivially tested, by allowing it take only strings and a writer as inputs. This means we no longer have to mock out parts of libmachine, and we can now compare outputs.

This PR introduces no behavior changes or new tests. A follow-up PR will fix the following bugs I discovered:

* If --unset and --no-proxy are used together, the output is corrupted
* cmd, powershell, and emacs add a trailing tab to the script
* cmd and powershell output contain an invalid newline
* unset output has incorrect comment

### CODE REVIEW

**Comment by tstromberg** [comment]
PTAL, I've repurposed this PR for cleanup-only work.

**Comment by medyagh** [comment]
Read half of it, looks very nice ! reading the rest

File: cmd/minikube/cmd/env.go [hunk 1]
```diff
@@ -361,43 +306,108 @@ var dockerEnvCmd = &cobra.Command{
...
+	tmpl := template.Must(template.New("envConfig").Parse(envTmpl))
+	return tmpl.Execute(w, shellCfgUnset(ec))
+}
+
+// dockerURL returns a the docker endpoint URL for an ip/port pair.
+func dockerURL(ip string, port int) string {
+	return fmt.Sprintf("tcp://%s", net.JoinHostPort(ip, strconv.Itoa(port)))
+}
+
+// dockerEnvVars gets the necessary docker env variables to allow the use of docker through minikube's vm
```
Comment (medyagh) [inline cmd/minikube/cmd/env.go]: *vm or container

**Pull request merged**